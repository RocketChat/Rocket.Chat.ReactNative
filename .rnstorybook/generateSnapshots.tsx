import React from 'react';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react-native';
import preview from './preview';
import { mockedStore } from '../app/reducers/mockedStore';
import { initStore } from '../app/lib/store/auxStore';

initStore(mockedStore);

export function generateSnapshots(stories: any) {
	describe('Story Snapshots:', () => {
		const composedStories = composeStories(stories, {
			decorators: preview.decorators
		});
		Object.entries(composedStories).forEach(([name, story]) => {
			test(`${name} should match snapshot`, () => {
				const rendered = render(React.createElement(story as React.ComponentType));
				expect(rendered.toJSON()).toMatchSnapshot();
			});
		});
	});
}
