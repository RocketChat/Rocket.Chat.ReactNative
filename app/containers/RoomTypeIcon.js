import React from 'react';
import { Image, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { CustomIcon } from '../lib/Icons';
import { themes } from '../constants/colors';

const styles = StyleSheet.create({
	style: {
		marginRight: 7,
		marginTop: 3
	},
	discussion: {
		marginRight: 6
	}
});

const RoomTypeIcon = React.memo(({
	type, size, style, theme
}) => {
	if (!type) {
		return null;
	}

	const color = themes[theme].auxiliaryText;

	if (type === 'discussion') {
		// FIXME: These are temporary only. We should have all room icons on <Customicon />, but our design team is still working on this.
		return <CustomIcon name='chat' size={13} style={[styles.style, styles.iconColor, styles.discussion, { color }]} />;
	}

	if (type === 'c') {
		return <Image source={{ uri: 'hashtag' }} style={[styles.style, style, { width: size, height: size, tintColor: color }]} />;
	} if (type === 'd') {
		return <CustomIcon name='at' size={13} style={[styles.style, styles.discussion, { color }]} />;
	} if (type === 'l') {
		return <CustomIcon name='livechat' size={13} style={[styles.style, styles.discussion, { color }]} />;
	}
	return <Image source={{ uri: 'lock' }} style={[styles.style, style, { width: size, height: size, tintColor: color }]} />;
});

RoomTypeIcon.propTypes = {
	theme: PropTypes.string,
	type: PropTypes.string,
	size: PropTypes.number,
	style: PropTypes.object
};

RoomTypeIcon.defaultProps = {
	size: 10
};

export default RoomTypeIcon;
