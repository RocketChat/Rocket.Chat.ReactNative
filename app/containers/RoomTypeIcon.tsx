import React from 'react';
import { StyleSheet, ViewStyle } from 'react-native';

import { CustomIcon } from '../lib/Icons';
import { STATUS_COLORS, themes } from '../constants/colors';
import Status from './Status/Status';
import { withTheme } from '../theme';

const styles = StyleSheet.create({
	icon: {
		marginRight: 4
	}
});

interface IRoomTypeIcon {
	theme?: string;
	type: string;
	isGroupChat?: boolean;
	teamMain?: boolean;
	status?: string;
	size?: number;
	style?: ViewStyle;
}

const RoomTypeIcon = React.memo(({ type, isGroupChat, status, style, theme, teamMain, size = 16 }: IRoomTypeIcon) => {
	if (!type) {
		return null;
	}

	const color = themes[theme!].titleText;
	const iconStyle = [styles.icon, { color }, style];

	if (type === 'd' && !isGroupChat) {
		return (
			<Status style={[iconStyle, { color: STATUS_COLORS[status!] ?? STATUS_COLORS.offline }]} size={size} status={status!} />
		);
	}

	// TODO: move this to a separate function
	let icon = 'channel-private';
	if (teamMain) {
		icon = `teams${type === 'p' ? '-private' : ''}`;
	} else if (type === 'discussion') {
		icon = 'discussions';
	} else if (type === 'c') {
		icon = 'channel-public';
	} else if (type === 'd') {
		if (isGroupChat) {
			icon = 'message';
		} else {
			icon = 'mention';
		}
	} else if (type === 'l') {
		icon = 'omnichannel';
	}

	return <CustomIcon name={icon} size={size} style={iconStyle} />;
});

export default withTheme(RoomTypeIcon);
