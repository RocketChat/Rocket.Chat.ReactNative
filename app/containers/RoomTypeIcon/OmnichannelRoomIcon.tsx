import React, { useState } from 'react';
import { StyleProp, ViewStyle } from 'react-native';
import { SvgUri } from 'react-native-svg';

import { OmnichannelSourceType, IOmnichannelSource } from '../../definitions';
import { STATUS_COLORS } from '../../lib/constants';
import { useAppSelector } from '../../lib/hooks';
import { CustomIcon, TIconsName } from '../CustomIcon';

interface IIconMap {
	[key: string]: TIconsName;
}

const iconMap: IIconMap = {
	widget: 'livechat-monochromatic',
	email: 'mail',
	sms: 'sms',
	app: 'omnichannel',
	api: 'omnichannel',
	other: 'omnichannel'
};

interface IOmnichannelRoomIconProps {
	size: number;
	type: string;
	style?: StyleProp<ViewStyle>;
	status?: string;
	sourceType?: IOmnichannelSource;
}

export const OmnichannelRoomIcon = ({ size, style, sourceType, status }: IOmnichannelRoomIconProps) => {
	const [loading, setLoading] = useState(true);
	const [svgError, setSvgError] = useState(false);
	const baseUrl = useAppSelector(state => state.server?.server);
	const connected = useAppSelector(state => state.meteor?.connected);

	const customIcon = (
		<CustomIcon
			name={iconMap[sourceType?.type || 'other']}
			size={size}
			style={style}
			color={STATUS_COLORS[status || 'offline']}
		/>
	);

	if (!svgError && sourceType?.type === OmnichannelSourceType.APP && sourceType.id && sourceType.sidebarIcon && connected) {
		return (
			<>
				<SvgUri
					height={size}
					width={size}
					color={STATUS_COLORS[status || 'offline']}
					uri={`${baseUrl}/api/apps/public/${sourceType.id}/get-sidebar-icon?icon=${sourceType.sidebarIcon}`}
					style={style}
					onError={() => setSvgError(true)}
					onLoad={() => setLoading(false)}
				/>
				{loading ? customIcon : null}
			</>
		);
	}

	return customIcon;
};
