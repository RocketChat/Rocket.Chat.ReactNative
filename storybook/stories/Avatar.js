import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import { themes } from '../../app/constants/colors';
import Avatar from '../../app/containers/Avatar/Avatar';
import Status from '../../app/containers/Status/Status';
import StoriesSeparator from './StoriesSeparator';
import sharedStyles from '../../app/views/Styles';

const styles = StyleSheet.create({
	status: {
		borderWidth: 4,
		bottom: -4,
		right: -4
	},
	custom: {
		padding: 16
	}
});

const server = 'https://open.rocket.chat';

const Separator = ({ title, theme }) => <StoriesSeparator title={title} theme={theme} />;
Separator.propTypes = {
	title: PropTypes.string,
	theme: PropTypes.string
};

const AvatarStories = ({ theme }) => (
	<ScrollView style={{ backgroundColor: themes[theme].backgroundColor }}>
		<Separator title='Avatar by text' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by roomId' theme={theme} />
		<Avatar
			type='p'
			rid='devWBbYr7inwupPqK'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by url' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by path' theme={theme} />
		<Avatar
			avatar='/avatar/diego.mello'
			server={server}
			size={56}
		/>
		<Separator title='With ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			avatarETag='5ag8KffJcZj9m5rCv'
			server={server}
			size={56}
		/>
		<Separator title='Without ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			server={server}
			size={56}
		/>
		<Separator title='Emoji' theme={theme} />
		<Avatar
			emoji='troll'
			getCustomEmoji={() => ({ name: 'troll', extension: 'jpg' })}
			server={server}
			size={56}
		/>
		<Separator title='Direct' theme={theme} />
		<Avatar
			text='diego.mello'
			server={server}
			type='d'
			size={56}
		/>
		<Separator title='Channel' theme={theme} />
		<Avatar
			text='general'
			server={server}
			type='c'
			size={56}
		/>
		<Separator title='Touchable' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			onPress={() => console.log('Pressed!')}
			size={56}
		/>
		<Separator title='Static' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			isStatic
			size={56}
		/>
		<Separator title='Custom borderRadius' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			borderRadius={28}
			size={56}
		/>
		<Separator title='Children' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		>
			<Status
				size={24}
				style={[sharedStyles.status, styles.status]}
				theme={theme}
			/>
		</Avatar>
		<Separator title='Wrong server' theme={theme} />
		<Avatar
			text='Avatar'
			server='https://google.com'
			size={56}
		/>
		<Separator title='Custom style' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
			style={styles.custom}
		/>
	</ScrollView>
);
AvatarStories.propTypes = {
	theme: PropTypes.string
};
export default AvatarStories;
