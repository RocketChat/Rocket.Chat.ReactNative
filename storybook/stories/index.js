/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { storiesOf } from '@storybook/react-native';

import RoomItem from './RoomItem';
import Message from './Message';
import UiKitMessage from './UiKitMessage';
import UiKitModal from './UiKitModal';
import Markdown from './Markdown';
// import RoomViewHeader from './RoomViewHeader';

import MessageContext from '../../app/containers/message/Context';

// MessageProvider
const baseUrl = 'https://open.rocket.chat';
const user = {
	id: '',
	username: 'diego.mello',
	token: ''
};

// Change here to see themed storybook
const theme = 'light';

const reducers = combineReducers({
	settings: () => ({}),
	login: () => ({
		user: {
			username: 'diego.mello'
		}
	}),
	meteor: () => ({ connected: true }),
	activeUsers: () => ({ abc: { status: 'online', statusText: 'dog' } })
});
const store = createStore(reducers);

const messageDecorator = story => (
	<MessageContext.Provider
		value={{
			user,
			baseUrl,
			onPress: () => {},
			onLongPress: () => {},
			reactionInit: () => {},
			onErrorPress: () => {},
			replyBroadcast: () => {},
			onReactionPress: () => {},
			onDiscussionPress: () => {},
			onReactionLongPress: () => {}
		}}
	>
		{story()}
	</MessageContext.Provider>
);

storiesOf('RoomItem', module)
	.addDecorator(story => <Provider store={store}>{story()}</Provider>)
	.add('list roomitem', () => <RoomItem theme={theme} />);
storiesOf('Message', module)
	.addDecorator(messageDecorator)
	.add('list message', () => <Message theme={theme} />);

storiesOf('UiKitMessage', module)
	.addDecorator(messageDecorator)
	.add('list uikitmessage', () => <UiKitMessage theme={theme} />);
storiesOf('UiKitModal', module)
	.addDecorator(messageDecorator)
	.add('list UiKitModal', () => <UiKitModal theme={theme} />);
storiesOf('Markdown', module)
	.add('list Markdown', () => <Markdown theme={theme} />);

// FIXME: I couldn't make these pass on jest :(
// storiesOf('RoomViewHeader', module)
// 	.add('list', () => <RoomViewHeader theme='black' />);
