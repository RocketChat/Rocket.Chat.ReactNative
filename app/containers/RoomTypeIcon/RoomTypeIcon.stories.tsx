import React from 'react';

import { OmnichannelSourceType } from '../../definitions';
import RoomTypeIcon from '.';

export default {
	title: 'RoomTypeIcon'
};

export const All = () => (
	<>
		<RoomTypeIcon size={30} type='d' userId='asd' />
		<RoomTypeIcon size={30} type='d' userId='asd' status='online' />
		<RoomTypeIcon size={30} type='d' isGroupChat />
		<RoomTypeIcon size={30} type='c' />
		<RoomTypeIcon size={30} type='p' />
		<RoomTypeIcon size={30} type='c' teamMain />
		<RoomTypeIcon size={30} type='p' teamMain />
		<RoomTypeIcon size={30} type='discussion' />
		<RoomTypeIcon size={30} type='l' status='away' sourceType={{ type: OmnichannelSourceType.SMS }} />
		<RoomTypeIcon size={30} type='p' style={{ margin: 10 }} />
	</>
);
