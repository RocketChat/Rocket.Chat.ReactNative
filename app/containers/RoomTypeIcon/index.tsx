import React from 'react';
import { StyleSheet, ViewStyle } from 'react-native';

import { OmnichannelRoomIcon } from './OmnichannelRoomIcon';
import { CustomIcon, TIconsName } from '../CustomIcon';
import { STATUS_COLORS, themes } from '../../lib/constants';
import Status from '../Status/Status';
import { useTheme } from '../../theme';
import { TUserStatus, IOmnichannelSource } from '../../definitions';

const styles = StyleSheet.create({
	icon: {
		marginRight: 4
	}
});

interface IRoomTypeIcon {
	type: string;
	isGroupChat?: boolean;
	teamMain?: boolean;
	status?: TUserStatus;
	size?: number;
	style?: ViewStyle;
	sourceType?: IOmnichannelSource;
}

const RoomTypeIcon = React.memo(({ type, isGroupChat, status, style, teamMain, size = 16, sourceType }: IRoomTypeIcon) => {
	const { theme } = useTheme();

	if (!type) {
		return null;
	}

	const color = themes[theme].titleText;
	const iconStyle = [styles.icon, { color }, style];

	if (type === 'd' && !isGroupChat) {
		if (!status) {
			status = 'offline';
		}
		return <Status style={[iconStyle, { color: STATUS_COLORS[status] }]} size={size} status={status} />;
	}

	if (type === 'l') {
		return <OmnichannelRoomIcon style={[styles.icon, style]} size={size} type={type} status={status} sourceType={sourceType} />;
	}

	// TODO: move this to a separate function
	let icon: TIconsName = 'channel-private';
	if (teamMain) {
		icon = `teams${type === 'p' ? '-private' : ''}`;
	} else if (type === 'discussion') {
		icon = 'discussions';
	} else if (type === 'c') {
		icon = 'channel-public';
	} else if (type === 'd') {
		if (isGroupChat) {
			icon = 'message';
		} else {
			icon = 'mention';
		}
	}

	return <CustomIcon name={icon} size={size} color={color} style={iconStyle} />;
});

export default RoomTypeIcon;
