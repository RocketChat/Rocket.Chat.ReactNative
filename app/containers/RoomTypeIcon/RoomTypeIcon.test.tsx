import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './RoomTypeIcon.stories';

generateSnapshots(stories);
