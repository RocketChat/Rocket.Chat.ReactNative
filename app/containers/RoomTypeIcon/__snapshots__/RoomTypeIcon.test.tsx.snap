// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: All should match snapshot 1`] = `
[
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#6C727A",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "height": 60,
              "textAlignVertical": "center",
              "width": 60,
            },
            [
              {
                "marginRight": 4,
              },
              undefined,
            ],
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#158D65",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "height": 60,
              "textAlignVertical": "center",
              "width": 60,
            },
            [
              {
                "marginRight": 4,
              },
              undefined,
            ],
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#1F2329",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "marginRight": 4,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#1F2329",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "marginRight": 4,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#1F2329",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "marginRight": 4,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#1F2329",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "marginRight": 4,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#1F2329",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "marginRight": 4,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#1F2329",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "marginRight": 4,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#AC892F",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "marginRight": 4,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#1F2329",
          "fontSize": 60,
        },
        [
          {
            "lineHeight": 60,
          },
          [
            {
              "marginRight": 4,
            },
            {
              "margin": 10,
            },
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
]
`;
